### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by reshetniktati**

### CODE REVIEW

File: src/app/group/tabs/result-tab/result-tab.component.ts [hunk 1]
```diff
@@ -27,20 +27,20 @@ export class ResultTabComponent implements OnInit {
...
   .subscribe(user => this.currentUser = user);
+  }
 
-}
+  ngOnChanges() {
+    this.currentGroupId = this.route.snapshot.paramMap.get('groupId');
+
+    this.userService.getStudentsByGroupId(this.currentGroupId)
+    .subscribe(students => { this.students = students; this.fillModel(students) });
+    console.log(this.students);
```
Comment (georgelviv) [inline src/app/group/tabs/result-tab/result-tab.component.ts]: should we keep console.log ? and don't forget about spaces ;)

File: src/app/group/tabs/result-tab/result-tab.component.ts [hunk 2]
```diff
@@ -77,6 +77,7 @@ export class ResultTabComponent implements OnInit {
       });
     }
     this.model = newModel;
+
```
Comment (georgelviv) [inline src/app/group/tabs/result-tab/result-tab.component.ts]: empty line, makes any sense here?

File: src/app/shared/services/user.service.ts [hunk 3]
```diff
@@ -12,61 +14,75 @@ const key: string = 'users';
...
-        return of(students);
-    }
+  public getStudentsByGroupId(groupId: string): Observable<User[]> {
+      let students = this.users.filter(user => !user.isAdmin && user.groupId === groupId);
+      return of(students);
+  }
+
+  // public getStudentById(studentId: string): Observable<User> {
+  //     let student = this.users.filter(user => !user.isAdmin && user.id === studentId)[0];
+  //     return of(student);
```
Comment (georgelviv) [inline src/app/shared/services/user.service.ts]: commented code = redundant code

File: src/app/group/homework-tab/homework-tab.component.ts [hunk 4]
```diff
@@ -1,32 +1,50 @@
-import {Component, OnInit, Input} from '@angular/core';
+import {Component, Input} from '@angular/core';
 
-import {UserService} from "../../shared/services/user.service";
+import {UserService} from '../../shared/services/user.service';
+import {Homework} from '../../shared/interface/homework.interface';
```
[the remark this answers is not in the document]
Comment (reshetniktati) [reply src/app/group/homework-tab/homework-tab.component.ts]: what is an alias?

[the remark this answers is not in the document]
Comment (georgelviv) [reply src/app/group/homework-tab/homework-tab.component.ts]: https://medium.com/@tomastrajan/6-best-practices-pro-tips-for-angular-cli-better-developer-experience-7b328bc9db81 
2nd point

File: src/app/group/homework-tab/homework-tab.component.ts [hunk 5]
```diff
@@ -1,32 +1,50 @@
...
-        this.currentStudent = {};
     }
-
-    ngOnInit() {
-    };
+    
+
+    ngOnChanges() {
+        this._userService.getUserById('128736').subscribe(user => {
+                this.currentUser = user;
```
[the remark this answers is not in the document]
Comment (reshetniktati) [reply src/app/group/homework-tab/homework-tab.component.ts]: Fixed

File: src/app/group/homework-tab/homework-tab.component.ts [hunk 6]
```diff
@@ -1,32 +1,50 @@
...
+
+                for(let i = 0; i < this.currentUser.lessons.length; i++) {
+                    if (this.currentUser.lessons[i].lessonId === this.selectedTask.id) {
```
[the remark this answers is not in the document]
Comment (reshetniktati) [reply src/app/group/homework-tab/homework-tab.component.ts]: Fixed

File: src/app/group/tabs/result-tab/result-tab.component.ts (hunk 1 above)
Comment (reshetniktati) [reply src/app/group/tabs/result-tab/result-tab.component.ts]: fixed

File: src/app/group/tabs/result-tab/result-tab.component.ts (hunk 2 above)
Comment (reshetniktati) [reply src/app/group/tabs/result-tab/result-tab.component.ts]: fixed

File: src/app/shared/services/user.service.ts (hunk 3 above)
Comment (reshetniktati) [reply src/app/shared/services/user.service.ts]: fixed

**Pull request merged**